reg/drone/service/battery/Parameters.0.2.uavcan

```diff
@@ -0,0 +1,57 @@
+# Smart battery parameter message. It is mostly intended for automated battery charging and maintenance systems.
+# This message is modeled after the Smart Battery Data Specification (SBS) and the MAVLink battery status messages.
+#
+# The values carried by this message are either constant or slow-changing, so, generally, the publishing frequency
+# should not be higher than 0.2 Hz, and the priority should be either OPTIONAL or SLOW.
+#
+# All parameters are required unless specifically stated otherwise.
+
+truncated uint64 unique_id
+# A statistically unique number that can be used to differentiate this exact battery from others.
+# Needed for logging purposes. If the battery supports SBS, this value may be computed as CRC64WE of
+# (ManufacturerName + DeviceName + ManufactureDate + SerialNumber).
+# This value shall be invariant to the identity of the reporting node unless it is an integral part of the battery.
+# Vendors may use a constant random number for the 40 most significant bits for vendor identification purposes.
+
+uavcan.si.unit.mass.Scalar.1.0 mass
+# The total mass of the battery, including the packaging, electronics, cabling, and all auxiliary items, if any.
+# May be used for predicting the kinematic parameters of the vehicle.
+# NaN if unknown.
+
+uavcan.si.unit.electric_charge.Scalar.1.0 design_capacity
+# The maximum total charge of the pack, at 100% SoH, specified by the manufacturer.
+
+uavcan.si.unit.voltage.Scalar.1.0[2] design_cell_voltage_min_max
+# The minimum (end of discharge) and the maximum (end of charge) cell voltages specified by the manufacturer
+# at 100% SoH. The cell voltages are measured when they have stabilized after discharge/charge has been terminated. Example: {2.8, 4.2} V
```

Comment:
> The cell voltages are measured when they have stabilized after discharge/charge has been terminated.

Not sure this is relevant because these voltages are not intended to be measured at all; instead, they are *expected by design*. A typical LiCOO<sub>2</sub> battery would always report constant 2.8, 4.2 without measuring anything.

[The CI says](https://travis-ci.org/github/UAVCAN/public_regulated_data_types/builds/756606157#L439) that this line exceeds the 120-character length limit, so do some of the following lines. Please reformat the comments.

Reply:
The comment reads as
```
The minimum (end of discharge) and the maximum (end of charge) cell voltages specified by the manufacturer at 100% SoH. This voltage correspond to resting voltage when they have stabilized after discharge/charge has been terminated. Example: {2.8, 4.2} V
```
It is merely to mention to a user that
1. the voltages outside a min and max voltages may be observed during use. 
2. min and max voltages are resting voltages

Reply:
The updated text reads better but it adds little information. I would omit it but if you strongly prefer otherwise feel free to update it as suggested (the grammar needs some tweaking as well).

Reply:
I'd strongly suggest to keep it that way. What is the grammar change ?

Reply:
@clyde-johnston would it be appropriate to summon your grammatical powers here?

Reply:
> The minimum (end of discharge) and the maximum (end of charge) cell voltages specified by the manufacturer at 100% SoH. This voltage correspond to resting voltage when they have stabilized after discharge/charge has been terminated. Example: {2.8, 4.2} V

The second sentence is most problematic.  It should be in the plural if you are referring to both the minimum and maximum cell voltages.

> The minimum (end of discharge) and the maximum (end of charge) cell voltages specified by the manufacturer at 100% SoH. These voltages correspond to the resting voltages when they have stabilized after the discharge/charge has been terminated. Example: {2.8, 4.2} V

Also, it looks like the "when..." clause is just a definition of "the resting voltages" in which case I would suggest writing it as:

> The minimum (end of discharge) and the maximum (end of charge) cell voltages specified by the manufacturer at 100% SoH. These voltages correspond to resting voltages i.e. the stabilized voltages after the discharge/charge has been terminated. Example: {2.8, 4.2} V

Reply:
My english is worse than I imagined.